### ISSUE TITLE
Question on Registration Recall

### ISSUE THREAD

**Issue by qsisi** [WHU-USI3DV/CoFiI2P#18]
Thanks for open-sourcing this project!

I have a question related to the registration recall on KITTI, looks like in your paper, the recall is 100% on kitti:
![image](https://github.com/user-attachments/assets/f6aab300-b278-4685-9dca-0f292f85c84b)
but when I ran the evaluation on kitti, the angles_diff and t_diff printed out is something like:
![image](https://github.com/user-attachments/assets/f74a7b03-baee-4243-a666-71f109298470)
most of the printed information indicates the angle & translation diff are large, which is strange to me.
Could you provide some hints about it?

Thanks for your reply!

**Comment by martin-liao** [comment]
Did you load the pre-trained weight correctly?

**Comment by qsisi** [comment]
I think I've loaded the pre-trained model correctly, I only modified this line:
![image](https://github.com/user-attachments/assets/8cb6bd63-2a07-43b6-97e7-838cbe748170)
and here are the results:
![image](https://github.com/user-attachments/assets/b225df48-da12-4732-bc9d-1296c9604773)

**Comment by qsisi** [comment]
kitti train from scratch：
![image](https://github.com/user-attachments/assets/babbc896-c0df-42b5-a371-68f618b66a8d)
kitti finetuned on **cofii2p_kitti.t7**：
![image](https://github.com/user-attachments/assets/84e18739-a7c9-4897-8d61-25be837b2e03)
Looks like the released weight  **cofii2p_kitti** only converges on **loss_coarse**?

**Comment by qsisi** [comment]
Any updates over the last few days?

**Comment by martin-liao** [comment]
We have double-checked the evaluation code and everything is fine.
Could you please provide more details, like configurations?

**Comment by martin-liao** [comment]
> kitti train from scratch： ![image](https://private-user-images.githubusercontent.com/44374058/372012189-babbc896-c0df-42b5-a371-68f618b66a8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxNDAxODUsIm5iZiI6MTcyODEzOTg4NSwicGF0aCI6Ii80NDM3NDA1OC8zNzIwMTIxODktYmFiYmM4OTYtYzBkZi00MmI1LWEzNzEtNjhmNjE4YjY2YThkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDE0NTEyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE5ZTE1NzE4MTM4M2VmMzdkNTMwMzY3ODc5NTY4NmVmYTQ5MmYzODhhNTM4MGYwZDhmZmZkN2E2ZThmMWQ1N2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.oaIApus2zYyWayWM5zFBDoKznp35V5oVLLFN567RVfw) kitti finetuned on **cofii2p_kitti.t7**： ![image](https://private-user-images.githubusercontent.com/44374058/372012445-84e18739-a7c9-4897-8d61-25be837b2e03.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgxNDAxODUsIm5iZiI6MTcyODEzOTg4NSwicGF0aCI6Ii80NDM3NDA1OC8zNzIwMTI0NDUtODRlMTg3MzktYTdjOS00ODk3LThkNjEtMjViZTgzN2IyZTAzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA1VDE0NTEyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTVjMTEzM2FhNTk2ZWJlZDdkYzk5MGFjMjFkYzEwYWM0NTAzYmU0NTU0ZDMwYmJiOTlhNTQ3NDA3ZmJiOGQ4M2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QQY4NC-Es2_MOqzMs52Ah1rH_rRw2OafUnbeAnDBH2E) Looks like the released weight **cofii2p_kitti** only converges on **loss_coarse**?
We have made further attempts and found that the evaluation results (RRE & RTE) are close to the values reported in the paper.
We also recommend using Tensorboard to visualize the training details.

**Comment by qsisi** [comment]
After re-installing the cuda、torch、open3d as required, the problem is solved.

**Comment by qsisi** [comment]
Hello! Sorry to bother you again, but my question is what is the purpose to normalize the query features here?

https://github.com/WHU-USI3DV/CoFiI2P/blob/main/model/transformer/transformer.py#L53

Thanks for your reply.

**Comment by martin-liao** [comment]
It's just an implementation trick. 
We found that normalized query vector leads to better performance.
Other variants may work, while you can have a try.
